### DISCUSSION TITLE
vscode TIMELINE

### DISCUSSION THREAD

**Discussion by 0x2CA** [zed-industries/zed#24673] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

There's nothing scarier in code development than deleting a file by mistake. Sometimes it can be a pop-up box, a wrong button, or a git error, and the fruits of days of hard work are gone.

It would be nice to have a feature that could record every version of your saved code locally and retrieve it at any time, so you wouldn't have to worry about losing files by mistake.

In VSCode, this used to be done by installing plugins (such as the Local History plugin), but after version 1.44, VSCode has a built-in Timeline feature.

**Comment by KTibow** [comment]
Also see: https://github.com/zed-industries/zed/discussions/24004
